feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/assets/search/AssetSearchInteractor.kt

```diff
@@ -52,6 +57,29 @@ class AssetSearchInteractor(
         }
     }
 
+    suspend fun searchSwapAssetsFlow(
+        forAsset: FullChainAssetId?,
+        queryFlow: Flow<String>,
+        externalBalancesFlow: Flow<List<ExternalBalance>>,
+        coroutineScope: CoroutineScope
+    ): Flow<Map<AssetGroup, List<AssetWithOffChainBalance>>> {
+        val availableAssets = getAvailableSwapAssets(forAsset, coroutineScope)
+
+        return searchAssetsInternalFlow(queryFlow, externalBalancesFlow) {
+            val chainAsset = it.token.configuration
+            availableAssets.contains(FullChainAssetId(chainAsset.chainId, chainAsset.id))
+        }
+    }
+
+    private suspend fun getAvailableSwapAssets(asset: FullChainAssetId?, coroutineScope: CoroutineScope): Set<FullChainAssetId> {
+        val chainAsset = asset?.let { chainRegistry.asset(it) }
+        return if (chainAsset == null) {
+            swapService.assetsAvailableForSwap(coroutineScope)
+        } else {
+            swapService.availableSwapDirectionsFor(chainAsset, coroutineScope)
+        }
+    }
```

Comment:
Is there some delay here before `swapService` will load all necessary information? If so maybe we should consult with Andrey about loading state

Reply:
Yes we have a delay here while assets are loading for swap flow